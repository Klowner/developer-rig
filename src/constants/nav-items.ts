import { ExtensionMode } from './extension-coordinator';

export const Labels: { [key: string]: string } = {
  ExtensionViews: 'Extension Views',
  ProductManagement: 'Product Management',
};
