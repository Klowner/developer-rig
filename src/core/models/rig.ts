import * as React from 'react';
import { PositionProperty } from 'csstype';

export interface RigExtensionView {
  x: number;
  y: number;
  orientation: string;
  id: string;
  channelId: string;
  extension: ExtensionCoordinator.ExtensionObject;
  features: {
    isChatEnabled: boolean;
  };
  type: string;
  mode?: string;
  role: string;
  linked: boolean;
  isPopout: boolean;
  deleteViewHandler?: (id: string) => void;
  openEditViewHandler?: (id: string) => void;
  frameSize?: FrameSize;
}

export interface FrameSize {
  height: number;
  width: number;
}
