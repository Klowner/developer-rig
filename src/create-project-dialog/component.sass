@import "../dialog-component/component.sass"
.project-dialog


.project-dialog__background
  @extend .dialog__backdrop

.project-dialog__dialog
  @extend .dialog
  height: 400px

.project-dialog__content
  @extend .container__left-block
  white-space: pre
  overflow-y: auto
  height: 250px
  width: 95%

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
