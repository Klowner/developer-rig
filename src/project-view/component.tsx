import * as React from 'react';
import './component.sass';
import { RigProject } from '../core/models/rig';

type ProjectViewProps = RigProject & {
  edit: Function;
};

export class ProjectView extends React.Component<ProjectViewProps> {
  edit = () => {
    this.props.edit();
  }

  public render() {
    return (
      <div className="project-view__content">
        <div className="project-view__state-value">
          <label className="props-value__label"> Is Local </label>
          <input disabled={true} className="props-value__input" type="checkbox" name="isLocal" checked={this.props.isLocal} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Name </label>
          <input disabled={true} className="props-value__input" type="text" name="name" value={this.props.name} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Owner Name </label>
          <input disabled={true} className="props-value__input" type="text" name="ownerName" value={this.props.ownerName} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Client ID </label>
          <input disabled={true} className="props-value__input" type="text" name="clientId" value={this.props.clientId} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Version </label>
          <input disabled={true} className="props-value__input" type="text" name="version" value={this.props.version} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Front-end Directory </label>
          <input disabled={true} className="props-value__input" type="text" name="frontend" value={this.props.frontend} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Back-end File </label>
          <input disabled={true} className="props-value__input" type="text" name="backend" value={this.props.backend} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Secret </label>
          <input disabled={true} className="props-value__input" type="text" name="secret" value={this.props.secret} />
        </div>
        <div className="project-view__state-value">
          <label className="props-value__label"> Manifest </label>
          <textarea disabled={true} className="props-value__input" name="manifest" value={JSON.stringify(this.props.manifest)} />
        </div>
        <div className="project-view__state-value">
          <button onClick={this.edit}>Edit</button>
        </div>
      </div>
    );
  }
}
