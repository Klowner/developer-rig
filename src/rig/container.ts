import { connect, Dispatch } from 'react-redux';
import { RigComponent, ReduxDispatchProps, ReduxStateProps } from './component';
import * as userActions from '../core/actions/user-session';
import { GlobalState } from '../core/models/global-state';
import { getUserSession } from '../core/state/session';

function mapStateToProps(state: GlobalState): ReduxStateProps {
  return {
    session: getUserSession(state),
  }
}

function mapDispatchToProps(dispatch: Dispatch<GlobalState>): ReduxDispatchProps {
  return {
    userLogin: userSession => dispatch(userActions.userLogin(userSession))
  };
}

export const Rig = connect<ReduxStateProps, ReduxDispatchProps>(mapStateToProps, mapDispatchToProps)(RigComponent);
