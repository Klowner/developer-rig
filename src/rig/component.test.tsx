import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { createViewsForTest, ExtensionForTest } from '../tests/constants/extension';
import { mockFetchForManifest, mockFetchForUserInfo } from '../tests/mocks';
import { Labels } from '../constants/nav-items';
import { RigComponent } from './component';
import { ExtensionAnchors } from '../constants/extension-types';
import { ViewerTypes } from '../constants/viewer-types';
import { RigExtensionView } from '../core/models/rig';
import { ExtensionViewDialog, ExtensionViewDialogState } from '../extension-view-dialog';

import { ExtensionAnchor, ExtensionMode, ExtensionViewType } from '../constants/extension-coordinator';

let globalAny = global as any;

describe('<RigComponent />', () => {
  const setupShallow = setupShallowTest(RigComponent, () => ({}));

  function setupViewsForTest(numViews: number) {
    const testViews = createViewsForTest(numViews, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    localStorage.setItem('extensionViews', JSON.stringify(testViews));
  };

  function setupComponentViewsForTest(numViews: number) {
    const testViews = createViewsForTest(numViews, ExtensionAnchors[ExtensionAnchor.Component], ViewerTypes.LoggedOut);
    localStorage.setItem('extensionViews', JSON.stringify(testViews));
  };

  beforeEach(() => {
    globalAny.fetch = jest.fn().mockImplementation(mockFetchForManifest);
  })

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('gets extension views from local storage correctly when not in local storage', () => {
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;
    expect(instance.getExtensionViews()).toEqual([]);
  });

  it('renders extension view correctly', () => {
    localStorage.setItem('projects', JSON.stringify([{}]));
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
    expect((wrapper.find('ExtensionViewContainer') as any).props().extensionViews).toHaveLength(1);
  });

  it('gets extension views from local storage correctly', () => {
    setupViewsForTest(1);
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;
    expect(instance.getExtensionViews()).toEqual(testViews);
  });

  it('deletes extension view correctly', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;
    expect((wrapper.find('ExtensionViewContainer') as any).props().extensionViews).toHaveLength(1);

    instance.deleteExtensionView('1');
    wrapper.update();
    expect((wrapper.find('ExtensionViewContainer') as any).props().extensionViews).toHaveLength(0);
  });

  it('toggles state when edit dialog is opened/closed', () => {
    setupComponentViewsForTest(1);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;

    instance.openEditViewHandler('1');
    expect(instance.state.showEditView).toBe(true);
    expect(instance.state.idToEdit).toBe('1');

    instance.closeEditViewHandler();
    expect(instance.state.showEditView).toBe(false);
    expect(instance.state.idToEdit).toBe('0');
  });

  it('edit changes the view and sets them correctly', () => {
    setupComponentViewsForTest(1);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;

    instance.openEditViewHandler('1');
    expect(instance.state.showEditView).toBe(true);
    expect(instance.state.idToEdit).toBe('1');

    instance.editViewHandler({ x: 25, y: 25 });

    const views = instance.getExtensionViews();
    const editedView = views.filter((element: RigExtensionView) => element.id === '1');
    expect(editedView[0].x).toEqual(25);
    expect(editedView[0].y).toEqual(25);
    expect(instance.state.showEditView).toBe(false);
    expect(instance.state.idToEdit).toBe('0');
  });

  it('correctly toggles state when create extension view is opened/closed', () => {
    localStorage.setItem('projects', JSON.stringify([{ manifest: { views: [{}] } }]));
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    wrapper.setState({
      manifest: ExtensionForTest,
    })
    const instance = wrapper.instance() as RigComponent;
    instance.openExtensionViewHandler();
    expect(instance.state.showExtensionsView).toBe(true);


    instance.closeExtensionViewDialog();
    expect(instance.state.showExtensionsView).toBe(false);
  });

  it('correctly sets state when viewHandler is invoked', () => {
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;

    instance.viewerHandler(Labels.ExtensionViews);
    expect(instance.state.selectedView).toBe(Labels.ExtensionViews);
  });

  it('gets the correct views when getExtensionViews invoked', () => {
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;

    expect(instance.state.extensionViews).toEqual(testViews);
  });

  it('sets views in local storage correctly when pushExtensionViews invoked', () => {
    const testViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    setupViewsForTest(1);
    const { wrapper } = setupShallow();
    const instance = wrapper.instance() as RigComponent;

    expect(instance.state.extensionViews).toEqual(testViews);

    instance.state.extensionViews.push(createViewsForTest(1,ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut)[0] as RigExtensionView);
    instance.pushExtensionViews(instance.state.extensionViews);
  });

  describe('gets frame size from dialog ref correctly', () => {
    it('returns correct data for mobile ', () => {
      const { wrapper } = setupShallow();
      const testDialogState = {
        width: 0,
        height: 0,
        frameSize: 'iPhone X (375x822)',
        extensionViewType: ExtensionViewType.Mobile
      } as ExtensionViewDialogState;
      const expectedMobileFrameSize = {
        width: 375,
        height: 822,
      };
      const instance = wrapper.instance() as RigComponent;

      let frameSize = instance.getFrameSizeFromDialog(testDialogState);
      expect(frameSize).toEqual(expectedMobileFrameSize);
    });

    it('returns correct data for other types', () => {
      const { wrapper } = setupShallow();
      const overlayTestDialogState = {
        width: 0,
        height: 0,
        frameSize: '640x480',
        extensionViewType: ExtensionViewType.VideoOverlay
      } as ExtensionViewDialogState;
      const expectedOverlayFrameSize = {
        width: 640,
        height: 480,
      };
      const instance = wrapper.instance() as RigComponent;
      const frameSize = instance.getFrameSizeFromDialog(overlayTestDialogState);
      expect(frameSize).toEqual(expectedOverlayFrameSize);
    });

    it('returns correct data for custom size', () => {
      const { wrapper } = setupShallow();
      const overlayTestDialogState = {
        width: 100,
        height: 100,
        frameSize: 'Custom',
        extensionViewType: ExtensionViewType.VideoOverlay
      } as ExtensionViewDialogState;
      const expectedOverlayFrameSize = {
        width: 100,
        height: 100,
      };
      const instance = wrapper.instance() as RigComponent;
      const frameSize = instance.getFrameSizeFromDialog(overlayTestDialogState);
      expect(frameSize).toEqual(expectedOverlayFrameSize);
    });

    it('correctly fetches user info if login not in localStorage', () => {
      globalAny.fetch = jest.fn().mockImplementation(mockFetchForUserInfo);
      globalAny.window.location.hash = 'access_token=test&';

      const { wrapper } = setupShallow();
      expect(globalAny.fetch).toHaveBeenCalled();
    });
  });
});
