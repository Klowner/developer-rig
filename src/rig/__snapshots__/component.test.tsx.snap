// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RigComponent /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigComponent
    saveManifest={[Function]}
    session={
        Object {
            "authToken": "test",
            "displayName": "test",
            "id": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
    userLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(RigNavComponent)
          error={undefined}
          openConfigurationsHandler={[Function]}
          openProductManagementHandler={[Function]}
          selectedView="Extension Views"
          viewerHandler={[Function]}
/>,
        <div>
          <ExtensionViewContainer
                    deleteExtensionViewHandler={[Function]}
                    extensionViews={Array []}
                    openEditViewHandler={[Function]}
                    openExtensionViewHandler={[Function]}
          />
          <Console />
</div>,
      ],
      "className": "rig-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": undefined,
          "openConfigurationsHandler": [Function],
          "openProductManagementHandler": [Function],
          "selectedView": "Extension Views",
          "viewerHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ExtensionViewContainer
              deleteExtensionViewHandler={[Function]}
              extensionViews={Array []}
              openEditViewHandler={[Function]}
              openExtensionViewHandler={[Function]}
/>,
            false,
            false,
            false,
            false,
            <Console />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "deleteExtensionViewHandler": [Function],
              "extensionViews": Array [],
              "openEditViewHandler": [Function],
              "openExtensionViewHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(RigNavComponent)
            error={undefined}
            openConfigurationsHandler={[Function]}
            openProductManagementHandler={[Function]}
            selectedView="Extension Views"
            viewerHandler={[Function]}
/>,
          <div>
            <ExtensionViewContainer
                        deleteExtensionViewHandler={[Function]}
                        extensionViews={Array []}
                        openEditViewHandler={[Function]}
                        openExtensionViewHandler={[Function]}
            />
            <Console />
</div>,
        ],
        "className": "rig-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": undefined,
            "openConfigurationsHandler": [Function],
            "openProductManagementHandler": [Function],
            "selectedView": "Extension Views",
            "viewerHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ExtensionViewContainer
                deleteExtensionViewHandler={[Function]}
                extensionViews={Array []}
                openEditViewHandler={[Function]}
                openExtensionViewHandler={[Function]}
/>,
              false,
              false,
              false,
              false,
              <Console />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "deleteExtensionViewHandler": [Function],
                "extensionViews": Array [],
                "openEditViewHandler": [Function],
                "openExtensionViewHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RigComponent /> renders extension view correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RigComponent
    saveManifest={[Function]}
    session={
        Object {
            "authToken": "test",
            "displayName": "test",
            "id": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
    userLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(RigNavComponent)
          error={undefined}
          openConfigurationsHandler={[Function]}
          openProductManagementHandler={[Function]}
          selectedView="Extension Views"
          viewerHandler={[Function]}
/>,
        <div>
          <ExtensionViewContainer
                    deleteExtensionViewHandler={[Function]}
                    extensionViews={
                              Array [
                                        Object {
                                          "channelId": "twitch",
                                          "extension": Object {
                                            "authorName": "test",
                                            "bitsEnabled": false,
                                            "clientId": "mockClientId",
                                            "description": "description",
                                            "iconUrl": "iconUrl",
                                            "id": "id",
                                            "name": "name",
                                            "requestIdentityLink": false,
                                            "sku": "sku",
                                            "state": "Released",
                                            "summary": "summary",
                                            "token": "token",
                                            "vendorCode": "vendorCode",
                                            "version": "0.1",
                                            "views": Object {
                                              "component": Object {
                                                "aspectHeight": 3000,
                                                "aspectWidth": 2500,
                                                "canLinkExternalContent": false,
                                                "viewerUrl": "test",
                                                "zoom": false,
                                                "zoomPixels": 100,
                                              },
                                              "config": Object {
                                                "canLinkExternalContent": false,
                                                "viewerUrl": "test",
                                              },
                                              "liveConfig": Object {
                                                "canLinkExternalContent": false,
                                                "viewerUrl": "test",
                                              },
                                              "mobile": Object {
                                                "viewerUrl": "test",
                                              },
                                              "panel": Object {
                                                "canLinkExternalContent": false,
                                                "height": 300,
                                                "viewerUrl": "test",
                                              },
                                              "videoOverlay": Object {
                                                "canLinkExternalContent": false,
                                                "viewerUrl": "test",
                                              },
                                            },
                                            "whitelistedConfigUrls": Array [
                                              "foo",
                                            ],
                                            "whitelistedPanelUrls": Array [
                                              "bar",
                                            ],
                                          },
                                          "features": Object {
                                            "isChatEnabled": true,
                                          },
                                          "id": "1",
                                          "isPopout": false,
                                          "linked": false,
                                          "mode": "viewer",
                                          "orientation": "Portrait",
                                          "role": "Logged-Out Viewer",
                                          "type": "Panel",
                                          "x": 0,
                                          "y": 0,
                                        },
                                      ]
                    }
                    openEditViewHandler={[Function]}
                    openExtensionViewHandler={[Function]}
          />
          <Console />
</div>,
      ],
      "className": "rig-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": undefined,
          "openConfigurationsHandler": [Function],
          "openProductManagementHandler": [Function],
          "selectedView": "Extension Views",
          "viewerHandler": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ExtensionViewContainer
              deleteExtensionViewHandler={[Function]}
              extensionViews={
                            Array [
                                          Object {
                                            "channelId": "twitch",
                                            "extension": Object {
                                              "authorName": "test",
                                              "bitsEnabled": false,
                                              "clientId": "mockClientId",
                                              "description": "description",
                                              "iconUrl": "iconUrl",
                                              "id": "id",
                                              "name": "name",
                                              "requestIdentityLink": false,
                                              "sku": "sku",
                                              "state": "Released",
                                              "summary": "summary",
                                              "token": "token",
                                              "vendorCode": "vendorCode",
                                              "version": "0.1",
                                              "views": Object {
                                                "component": Object {
                                                  "aspectHeight": 3000,
                                                  "aspectWidth": 2500,
                                                  "canLinkExternalContent": false,
                                                  "viewerUrl": "test",
                                                  "zoom": false,
                                                  "zoomPixels": 100,
                                                },
                                                "config": Object {
                                                  "canLinkExternalContent": false,
                                                  "viewerUrl": "test",
                                                },
                                                "liveConfig": Object {
                                                  "canLinkExternalContent": false,
                                                  "viewerUrl": "test",
                                                },
                                                "mobile": Object {
                                                  "viewerUrl": "test",
                                                },
                                                "panel": Object {
                                                  "canLinkExternalContent": false,
                                                  "height": 300,
                                                  "viewerUrl": "test",
                                                },
                                                "videoOverlay": Object {
                                                  "canLinkExternalContent": false,
                                                  "viewerUrl": "test",
                                                },
                                              },
                                              "whitelistedConfigUrls": Array [
                                                "foo",
                                              ],
                                              "whitelistedPanelUrls": Array [
                                                "bar",
                                              ],
                                            },
                                            "features": Object {
                                              "isChatEnabled": true,
                                            },
                                            "id": "1",
                                            "isPopout": false,
                                            "linked": false,
                                            "mode": "viewer",
                                            "orientation": "Portrait",
                                            "role": "Logged-Out Viewer",
                                            "type": "Panel",
                                            "x": 0,
                                            "y": 0,
                                          },
                                        ]
              }
              openEditViewHandler={[Function]}
              openExtensionViewHandler={[Function]}
/>,
            false,
            false,
            false,
            false,
            <Console />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "deleteExtensionViewHandler": [Function],
              "extensionViews": Array [
                Object {
                  "channelId": "twitch",
                  "extension": Object {
                    "authorName": "test",
                    "bitsEnabled": false,
                    "clientId": "mockClientId",
                    "description": "description",
                    "iconUrl": "iconUrl",
                    "id": "id",
                    "name": "name",
                    "requestIdentityLink": false,
                    "sku": "sku",
                    "state": "Released",
                    "summary": "summary",
                    "token": "token",
                    "vendorCode": "vendorCode",
                    "version": "0.1",
                    "views": Object {
                      "component": Object {
                        "aspectHeight": 3000,
                        "aspectWidth": 2500,
                        "canLinkExternalContent": false,
                        "viewerUrl": "test",
                        "zoom": false,
                        "zoomPixels": 100,
                      },
                      "config": Object {
                        "canLinkExternalContent": false,
                        "viewerUrl": "test",
                      },
                      "liveConfig": Object {
                        "canLinkExternalContent": false,
                        "viewerUrl": "test",
                      },
                      "mobile": Object {
                        "viewerUrl": "test",
                      },
                      "panel": Object {
                        "canLinkExternalContent": false,
                        "height": 300,
                        "viewerUrl": "test",
                      },
                      "videoOverlay": Object {
                        "canLinkExternalContent": false,
                        "viewerUrl": "test",
                      },
                    },
                    "whitelistedConfigUrls": Array [
                      "foo",
                    ],
                    "whitelistedPanelUrls": Array [
                      "bar",
                    ],
                  },
                  "features": Object {
                    "isChatEnabled": true,
                  },
                  "id": "1",
                  "isPopout": false,
                  "linked": false,
                  "mode": "viewer",
                  "orientation": "Portrait",
                  "role": "Logged-Out Viewer",
                  "type": "Panel",
                  "x": 0,
                  "y": 0,
                },
              ],
              "openEditViewHandler": [Function],
              "openExtensionViewHandler": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(RigNavComponent)
            error={undefined}
            openConfigurationsHandler={[Function]}
            openProductManagementHandler={[Function]}
            selectedView="Extension Views"
            viewerHandler={[Function]}
/>,
          <div>
            <ExtensionViewContainer
                        deleteExtensionViewHandler={[Function]}
                        extensionViews={
                                    Array [
                                                Object {
                                                  "channelId": "twitch",
                                                  "extension": Object {
                                                    "authorName": "test",
                                                    "bitsEnabled": false,
                                                    "clientId": "mockClientId",
                                                    "description": "description",
                                                    "iconUrl": "iconUrl",
                                                    "id": "id",
                                                    "name": "name",
                                                    "requestIdentityLink": false,
                                                    "sku": "sku",
                                                    "state": "Released",
                                                    "summary": "summary",
                                                    "token": "token",
                                                    "vendorCode": "vendorCode",
                                                    "version": "0.1",
                                                    "views": Object {
                                                      "component": Object {
                                                        "aspectHeight": 3000,
                                                        "aspectWidth": 2500,
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                        "zoom": false,
                                                        "zoomPixels": 100,
                                                      },
                                                      "config": Object {
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                      },
                                                      "liveConfig": Object {
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                      },
                                                      "mobile": Object {
                                                        "viewerUrl": "test",
                                                      },
                                                      "panel": Object {
                                                        "canLinkExternalContent": false,
                                                        "height": 300,
                                                        "viewerUrl": "test",
                                                      },
                                                      "videoOverlay": Object {
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                      },
                                                    },
                                                    "whitelistedConfigUrls": Array [
                                                      "foo",
                                                    ],
                                                    "whitelistedPanelUrls": Array [
                                                      "bar",
                                                    ],
                                                  },
                                                  "features": Object {
                                                    "isChatEnabled": true,
                                                  },
                                                  "id": "1",
                                                  "isPopout": false,
                                                  "linked": false,
                                                  "mode": "viewer",
                                                  "orientation": "Portrait",
                                                  "role": "Logged-Out Viewer",
                                                  "type": "Panel",
                                                  "x": 0,
                                                  "y": 0,
                                                },
                                              ]
                        }
                        openEditViewHandler={[Function]}
                        openExtensionViewHandler={[Function]}
            />
            <Console />
</div>,
        ],
        "className": "rig-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": undefined,
            "openConfigurationsHandler": [Function],
            "openProductManagementHandler": [Function],
            "selectedView": "Extension Views",
            "viewerHandler": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ExtensionViewContainer
                deleteExtensionViewHandler={[Function]}
                extensionViews={
                                Array [
                                                Object {
                                                  "channelId": "twitch",
                                                  "extension": Object {
                                                    "authorName": "test",
                                                    "bitsEnabled": false,
                                                    "clientId": "mockClientId",
                                                    "description": "description",
                                                    "iconUrl": "iconUrl",
                                                    "id": "id",
                                                    "name": "name",
                                                    "requestIdentityLink": false,
                                                    "sku": "sku",
                                                    "state": "Released",
                                                    "summary": "summary",
                                                    "token": "token",
                                                    "vendorCode": "vendorCode",
                                                    "version": "0.1",
                                                    "views": Object {
                                                      "component": Object {
                                                        "aspectHeight": 3000,
                                                        "aspectWidth": 2500,
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                        "zoom": false,
                                                        "zoomPixels": 100,
                                                      },
                                                      "config": Object {
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                      },
                                                      "liveConfig": Object {
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                      },
                                                      "mobile": Object {
                                                        "viewerUrl": "test",
                                                      },
                                                      "panel": Object {
                                                        "canLinkExternalContent": false,
                                                        "height": 300,
                                                        "viewerUrl": "test",
                                                      },
                                                      "videoOverlay": Object {
                                                        "canLinkExternalContent": false,
                                                        "viewerUrl": "test",
                                                      },
                                                    },
                                                    "whitelistedConfigUrls": Array [
                                                      "foo",
                                                    ],
                                                    "whitelistedPanelUrls": Array [
                                                      "bar",
                                                    ],
                                                  },
                                                  "features": Object {
                                                    "isChatEnabled": true,
                                                  },
                                                  "id": "1",
                                                  "isPopout": false,
                                                  "linked": false,
                                                  "mode": "viewer",
                                                  "orientation": "Portrait",
                                                  "role": "Logged-Out Viewer",
                                                  "type": "Panel",
                                                  "x": 0,
                                                  "y": 0,
                                                },
                                              ]
                }
                openEditViewHandler={[Function]}
                openExtensionViewHandler={[Function]}
/>,
              false,
              false,
              false,
              false,
              <Console />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "deleteExtensionViewHandler": [Function],
                "extensionViews": Array [
                  Object {
                    "channelId": "twitch",
                    "extension": Object {
                      "authorName": "test",
                      "bitsEnabled": false,
                      "clientId": "mockClientId",
                      "description": "description",
                      "iconUrl": "iconUrl",
                      "id": "id",
                      "name": "name",
                      "requestIdentityLink": false,
                      "sku": "sku",
                      "state": "Released",
                      "summary": "summary",
                      "token": "token",
                      "vendorCode": "vendorCode",
                      "version": "0.1",
                      "views": Object {
                        "component": Object {
                          "aspectHeight": 3000,
                          "aspectWidth": 2500,
                          "canLinkExternalContent": false,
                          "viewerUrl": "test",
                          "zoom": false,
                          "zoomPixels": 100,
                        },
                        "config": Object {
                          "canLinkExternalContent": false,
                          "viewerUrl": "test",
                        },
                        "liveConfig": Object {
                          "canLinkExternalContent": false,
                          "viewerUrl": "test",
                        },
                        "mobile": Object {
                          "viewerUrl": "test",
                        },
                        "panel": Object {
                          "canLinkExternalContent": false,
                          "height": 300,
                          "viewerUrl": "test",
                        },
                        "videoOverlay": Object {
                          "canLinkExternalContent": false,
                          "viewerUrl": "test",
                        },
                      },
                      "whitelistedConfigUrls": Array [
                        "foo",
                      ],
                      "whitelistedPanelUrls": Array [
                        "bar",
                      ],
                    },
                    "features": Object {
                      "isChatEnabled": true,
                    },
                    "id": "1",
                    "isPopout": false,
                    "linked": false,
                    "mode": "viewer",
                    "orientation": "Portrait",
                    "role": "Logged-Out Viewer",
                    "type": "Panel",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "openEditViewHandler": [Function],
                "openExtensionViewHandler": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
